packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.25 <0.7.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/access/roles/WhitelistAdminRole.sol";
+
+interface ERC20fraction {
+  function decimals() external view returns (uint8);
+}
+
+interface AggregatorFraction {
+  function decimals() external view returns (uint8);
+  function latestAnswer() external view returns (int256);
+  function latestTimestamp() external view returns (uint256);
+}
+
+
+/**
+ * @title ChainlinkConversionPath
+ *
+ * @notice ChainlinkConversionPath is a contract allowing to compute conversion from a path of Chainlink
+ */
+contract ChainlinkConversionPath is WhitelistAdminRole {
+  using SafeMath for uint256;
+
+  uint constant DECIMALS = 1e18;
+
+  // Mapping of Chainlink aggrefators (input currency => output currency => contract address)
+  // input & output currencies are the addresses of the ERC20 contracts OR the sha3("currency code")
+  mapping(address => mapping(address => address)) public allAggregators;
+
+  // rate must have been updated before the last... 10min
+  uint256 public maxTimestampDeltaAcceptable = 600;
+
+  // declare a new aggregator
+  event UpdateAggregator(address _input, address _output, address _aggregator);
+
+  /**
+    * @notice Update an aggregator
+    * @param _input address representing the input currency
+    * @param _output address representing the output currency
+    * @param _aggregator address of the aggregator contract
+  */
+  function updateAggregator(address _input, address _output, address _aggregator)
+    external
+    onlyWhitelistAdmin
+  {
+    allAggregators[_input][_output] = _aggregator;
+    emit UpdateAggregator(_input, _output, _aggregator);
+  }
+
+  /**
+    * @notice Update a list of aggregators
+    * @param _inputs list of addresses representing the input currencies
+    * @param _outputs list of addresses representing the output currencies
+    * @param _aggregators list of addresses of the aggregator contracts
+  */
+  function updateListAggregators(address[] calldata _inputs, address[] calldata _outputs, address[] calldata _aggregators)
+    external
+    onlyWhitelistAdmin
+  {
+    require(_inputs.length == _outputs.length, "arrays must have the same length");
+    require(_inputs.length == _aggregators.length, "arrays must have the same length");
+
+    // For every conversions of the path
+    for (uint i; i < _inputs.length; i++) {
+      allAggregators[_inputs[i]][_outputs[i]] = _aggregators[i];
+      emit UpdateAggregator(_inputs[i], _outputs[i], _aggregators[i]);
+    }
+  }
+
+  /**
+  * @notice Computes the conversion from an amount through a list of conversion
+  * @param _amountIn Amount to convert
+  * @param _path List of addresses representing the currencies for the conversions
+  * @return result the result after all the conversions
```

Comment:
missing doc about `oldestTimestampRate`

Reply:
btw, if `oldestTimestampRate` represents the timestamp of the oldest rate, the name should be `oldestRateTimestamp`.  If it is the Rate of the oldest timestamp, the name is correct.